mesonbuild/build.py

```diff
@@ -2328,6 +2328,13 @@ def process_kwargs(self, kwargs):
             else:
                 raise InvalidArguments(f'Invalid rust_crate_type "{rust_crate_type}": must be a string.')
             if rust_crate_type == 'proc-macro':
+                if 'native' not in kwargs:
+                    kwargs['native'] = True
+                if not kwargs.get('native', False):
+                    if self.environment.is_cross_build():
+                        raise InvalidArguments('Rust "proc-macro" crate type requires "native: true". This will fail in a cross-build.')
+                    else:
+                        mlog.warning('Rust "proc-macro" crate type requires "native: true". This will fail in a cross-build.')
```

Comment:
It does not require "native: true", that's the default in this case. I think it should always be an hard error, no need to downgrade to a warning if it's not a cross build. To happen, user must have explicitely wrote:
```meson
shared_library(..., rust_crate_type: 'proc-macros', native: false)
```
which is always wrong.

What about:
```suggestion
                    raise InvalidArguments('Rust "proc-macro" crate type does not allow "native: false".')
```